migrations/245_grading_method_split.sql

```diff
@@ -0,0 +1,23 @@
+-- Add new grading columns
+ALTER TABLE submissions 
+    ADD COLUMN IF NOT EXISTS grading_method_internal boolean DEFAULT True,
+    ADD COLUMN IF NOT EXISTS grading_method_external boolean DEFAULT False,
+    ADD COLUMN IF NOT EXISTS grading_method_manual boolean DEFAULT False;
+ALTER TABLE questions
+    ADD COLUMN IF NOT EXISTS grading_method_internal boolean DEFAULT True,
+    ADD COLUMN IF NOT EXISTS grading_method_external boolean DEFAULT False,
+    ADD COLUMN IF NOT EXISTS grading_method_manual boolean DEFAULT False;
+
+-- Update based on old values
+UPDATE submissions 
+    SET grading_method_internal = grading_method = 'Internal',
+        grading_method_external = grading_method = 'External',
+        grading_method_manual   = grading_method = 'Manual';
+UPDATE questions 
+    SET grading_method_internal = grading_method = 'Internal',
+        grading_method_external = grading_method = 'External',
+        grading_method_manual   = grading_method = 'Manual';
+
+-- Drop old grading method column
+ALTER TABLE submissions DROP COLUMN IF EXISTS grading_method;
+ALTER TABLE questions DROP COLUMN IF EXISTS grading_method;
```

Comment:
This is almost certainly going to cause problems when it comes time to deploy this change, as for a while we'll have some servers running old and new versions of this code simultaneously. Unfortunately, I think we'll need to keep the old `grading_method` column around for a while and remove it once no more servers are trying to interact with it. Alternatively, we could decide to accept some downtime to deploy this change and take PL completely offline during the migration. However, that would complicate deployments for folks who are running PL themselves.

Reply:
Deployment wise, makes sense. Can we feature flag this to help reduce drift?